instrumentation/reactor/reactor-netty/reactor-netty-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/reactornetty/v1_0/HttpClientConnectInstrumentation.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.reactornetty.v1_0;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.returns;
+
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import reactor.core.publisher.Mono;
+import reactor.netty.Connection;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.http.client.HttpClientConfig;
+import reactor.netty.http.client.HttpClientConfigBuddy;
+
+public class HttpClientConnectInstrumentation implements TypeInstrumentation {
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("reactor.netty.http.client.HttpClientConnect");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        named("connect").and(returns(named("reactor.core.publisher.Mono"))),
+        this.getClass().getName() + "$ConnectAdvice");
+  }
+
+  @SuppressWarnings("unused")
+  public static class ConnectAdvice {
+
+    @Advice.OnMethodExit(suppress = Throwable.class)
+    public static void onExit(
+        @Advice.Return(readOnly = false) Mono<? extends Connection> connection,
+        @Advice.This HttpClient httpClient) {
+
+      HttpClientConfig config = httpClient.configuration();
+      if (HttpClientConfigBuddy.hasDeferredConfig(config)) {
+        connection = HttpClientConfigBuddy.getConnector(config).apply(connection);
+      }
```

Comment:
can you add comment about the workaround?

when this is fixed upstream will this cause any issue? or is that what the new `compareAndSet` is guarding against?

Reply:
> can you add comment about the workaround?

👍 

> when this is fixed upstream will this cause any issue? or is that what the new `compareAndSet` is guarding against?

The new `compareAndSet` is a "just in case" thing, I'm hoping we'll disable this particular instrumentation when a new reactor version comes out.

Reply:
> I'm hoping we'll disable this particular instrumentation when a new reactor version comes out.

we might want to apply it still on old reactor netty versions

Reply:
Yes, definitely -- I meant, disable it for the versions that have that fix